ui/src/message_popup/author/question_page.jsx

```diff
@@ -0,0 +1,284 @@
+/* @flow weak */
+/* eslint no-console: "off" */
+import React from 'react';
+import _ from 'lodash';
+import VelocityTransitionGroup from 'velocity-react/velocity-transition-group';
+import 'velocity-animate/velocity.ui';
+
+import * as EditingComponents from './question_editing_components/editing_component.jsx';
+
+import {allQuestions} from '../questions.js';
+import {allStudents} from '../../data/virtual_school.js';
+import {learningObjectives} from '../../data/learning_objectives.js';
+import {indicators} from '../../data/indicators.js';
+import * as Routes from '../../routes.js';
+
+import AppBar from 'material-ui/AppBar';
+import IconButton from 'material-ui/IconButton';
+import RaisedButton from 'material-ui/RaisedButton';
+import Dialog from 'material-ui/Dialog';
+import FlatButton from 'material-ui/FlatButton';
+
+import ArrowBackIcon from 'material-ui/svg-icons/navigation/arrow-back';
+
+export default React.createClass({
+  displayName: 'QuestionPage',
+
+  propTypes: {
+    originalQuestion: React.PropTypes.object,
+    questionText: React.PropTypes.string,
+    students: React.PropTypes.array,
+    learningObjective: React.PropTypes.object,
+    indicator: React.PropTypes.object,
+    goodExamplesText: React.PropTypes.string,
+    badExamplesText: React.PropTypes.string
+  },
+
+  getDefaultProps(){
+    return ({
+      questionText: '',
+      students: [],
+      learningObjective: learningObjectives[0],
+      indicator: indicators[0],
+      goodExamplesText: "",
+      badExamplesText: ""
+    });
+
+  },
+  getInitialState() {
+    return ({
+      questionText: this.props.questionText,
+      students: this.props.students,
+      learningObjective: this.props.learningObjective,
+      indicator: this.props.indicator,
+      goodExamplesText: this.props.goodExamplesText,
+      badExamplesText: this.props.badExamplesText,
+      deleteConfirmationOpen: false
+    });
+  },
+
+  returnToQuestions(){
```

Comment:
I left this a few places, but could you do a pass and update the event handlers to have names that indicate what happened, something like `onReturnToQuestionsClicked`?